pandas/core/dtypes/cast.py

```diff
@@ -1026,3 +1027,28 @@ def find_common_type(types):
             return np.object
 
     return np.find_common_type(types, [])
+
+
+def maybe_cast_to_integer(arr, dtype):
+    """
+    Find a common data type among the given dtypes.
+
+    Parameters
+    ----------
+    arr : array
+    dtype : dtype
+
+    Returns
+    -------
+    integer or unsigned integer array (or raise if the dtype is incompatible)
+
+    """
+
+    if is_unsigned_integer_dtype(dtype) and (np.asarray(arr) < 0).any():
```

Comment:
these are ndarays, so you don't need the ``np.asarray(arr)`` (though you *can* do it but it would be before any checks (it doesn't copy so its fine).